src/main/java/hudson/plugins/ec2/EC2SpotSlave.java

```diff
@@ -214,15 +217,14 @@ public void terminate() {
                                         LOGGER.info("Terminated EC2 instance (terminated): " + instanceId);
                                     } catch (AmazonClientException e) {
                                         // Spot request is no longer valid
-                                        LOGGER.log(
-                                                Level.WARNING,
-                                                "Failed to terminate the Spot instance: " + instanceId,
-                                                e);
+                                        Functions.printStackTrace(
+                                                e,
+                                                listener.error("Failed to terminate the Spot instance: " + instanceId));
                                     }
                                 }
                             }
                         } catch (Exception e) {
-                            LOGGER.log(Level.WARNING, "Failed to remove agent: ", e);
+                            Functions.printStackTrace(e, listener.error("Failed to remove agent"));
                         } finally {
                             // Remove the instance even if deletion failed, otherwise it will hang around forever in
                             // the nodes page. One way for this to occur is that an instance was terminated
```

Comment:
Pay attention to https://github.com/jenkinsci/jenkins/blob/4248fc0ca6ce108364ed23208063ff20753f5e70/core/src/main/java/hudson/slaves/AbstractCloudSlave.java#L88-L94 which means that after switching from`terminate` to `_terminate` it is no longer your responsibility to `removeNode`.